src/java.base/share/classes/jdk/internal/classfile/impl/BufWriterImpl.java

```diff
@@ -111,7 +120,16 @@ public void writeFloat(float x) {
 
     @Override
     public void writeLong(long x) {
-        writeIntBytes(8, x);
+        reserveSpace(8);
+        elems[offset    ] = (byte) (x >> 56);
```

Comment:
Since elems is mutable, should we pull it to a local variable and write instead? Does C2 optimize this current code in compile logs?